src/coreclr/src/gc/gc.h

```diff
@@ -107,7 +107,24 @@ class IGCHeapInternal;
 
 /* misc defines */
 #define LARGE_OBJECT_SIZE ((size_t)(85000))
-#define max_generation 2
+
+enum generation_num
```

Comment:
>generation_num [](start = 5, length = 14)

generation_num might not be a good name because in GC code we already have locals named generation_num... perhaps call it something like gc_generation_num?